tests/integration_tests/data_steward/retraction/retract_utils_test.py

```diff
@@ -0,0 +1,45 @@
+# Python imports
+import os
+from unittest import TestCase
+
+# Third party imports
+import google.cloud.bigquery as gbq
+
+# Project imports
+import app_identity
+import retraction.retract_utils as ru
+
+
+class RetractUtilsTest(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = app_identity.get_application_id()
+        self.dataset_id = os.environ.get('UNIONED_DATASET_ID')
+        self.client = gbq.Client()
+
+    def test_is_labeled_deid(self):
+        dataset = self.client.get_dataset(
+            f'{self.project_id}.{self.dataset_id}')
```

Comment:
Question, why aren't you explicitly calling `create_dataset` here?  Example for circleci [here](https://github.com/all-of-us/curation/blob/develop/data_steward/ci/test_datasets.py#L64).  And when you create the dataset, why not create it with the labels you are testing against?  I'm assuming this pre-condition check is caused by a delay with these asynchronous BQ API calls.  
So you might consider:
1. setting these up as three different tests (functions) 
2. OR implementing a wait/check to make sure the function exists or the change has been completed.
2. including a `tearDown()` to remove the test only dataset (this must be done).

Reply:
I just understood why it could be happening. I called `get_dataset` just once and after updating it, used the same dataset that was retrieved before the update to make subsequent `update_dataset` calls, which was outdated by then. I resolved it by calling `get_dataset` before making each of the subsequent `update_dataset` calls. I was hesitant to use `create_dataset` just to change the label since creating/deleting datasets is more resource and time consuming than an update, but if this issue pops up again, I'll have to resort to that.